Framework/PythonInterface/plugins/algorithms/CalculateEfficiencyCorrection.py

```diff
@@ -0,0 +1,256 @@
+                 where :math:`\\lambda_{ref} =` 1.7982. XSectionType has no effect. Default=0.0")
+
+        self.declareProperty(
+            name='XSectionType',
+            defaultValue="AttenuationXSection",
+            validator=StringListValidator(['AttenuationXSection', 'TotalXSection']),
+            doc = 'Use either the absorption cross section (for monitor-type correction) or, \
+                   the total cross section (for transmission-type correction) in Alpha term. \
+                   Default=AttenuationXSection')
+
+    def validateInputs(self):
+        issues = dict()
+
+        if self.getProperty('InputWorkspace').isDefault and self.getProperty('WavelengthRange').isDefault:
+            issues['InputWorkspace'] = "Must select either InputWorkspace and WavelengthRange as input"
+            issues['WavelengthRange'] = "Must select either InputWorkspace and WavelengthRange as input"
+
+        if not self.getProperty('InputWorkspace').isDefault and not self.getProperty('WavelengthRange').isDefault:
+            issues['InputWorkspace'] = "Cannot select both InputWorkspace and WavelengthRange as input"
+            issues['WavelengthRange'] = "Cannot select both InputWorkspace and WavelengthRange as input"
+
+        if not self.getProperty('Density').isDefault:
+            if self.getProperty('ChemicalFormula').isDefault:
+                issues['ChemicalFormula'] = "Must specify the ChemicalFormula with Density"
+
+        if not self.getProperty('MeasuredEfficiency').isDefault:
+            if self.getProperty('ChemicalFormula').isDefault:
+                issues['ChemicalFormula'] = "Must specify the ChemicalFormula with MeasuredEfficiency"
+
+        if not self.getProperty('MeasuredEfficiency').isDefault:
+            if self.getProperty('ChemicalFormula').isDefault:
+                issues['ChemicalFormula'] = "Must specify the ChemicalFormula with MeasuredEfficiency"
+
+        if not self.getProperty('MeasuredEfficiency').isDefault and not self.getProperty('Density').isDefault:
+            issues['MeasuredEfficiency'] = "Cannot select both MeasuredEfficiency and Density as input"
+            issues['Density'] = "Cannot select both MeasuredEfficiency and Density as input"
+
+        if not self.getProperty('Alpha').isDefault and not self.getProperty('Density').isDefault:
+            issues['Alpha'] = "Cannot select both Alpha and Density as input"
+            issues['Density'] = "Cannot select both Alpha and Density as input"
+
+        if not self.getProperty('MeasuredEfficiency').isDefault and not self.getProperty('Alpha').isDefault:
+            issues['MeasuredEfficiency'] = "Cannot select both MeasuredEfficiency and Alpha as input"
+            issues['Alpha'] = "Cannot select both MeasuredEfficiency and Alpha as input"
+
+        return issues
+
+    def _setup(self):
+        self._input_ws = self.getPropertyValue('InputWorkspace')
```

Comment:
`getPropertyValue` does not work if the input workspace is not stored in the ADS:
```python
input_wksp = CreateSampleWorkspace(WorkspaceType="Event", Function="User Defined",
    UserDefinedFunction="name=ExpDecay, Height=100, Lifetime=4",
    Xmin=0.2, Xmax=4.0, BinWidth=0.01, XUnit="Wavelength",
    NumEvents=10000, NumBanks=1, BankPixelWidth=1, StoreInADS=False)

# Calculate the efficiency correction
correction_wksp = CalculateEfficiencyCorrection(InputWorkspace=input_wksp, Alpha=0.5, StoreInADS=False)
```
In the above case, InputWorkspace will not have a name, `getPropertyValue` returns an empty string and `CloneWorkspace` in `PyExec` raises an exception.